uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java

```diff
@@ -88,6 +88,17 @@ public void performDeviceRegistration(@NonNull final String username,
         );
     }
 
+    public void performDeviceRegistrationForFederatedUser(@NonNull final String username,
```

Comment:
Why wouldn't we just add the following to `ITestBroker` and just override it here:

```java
    void performDeviceRegistration(String username, String password, boolean isFederated);
```

I don't think we need a method specifically for only federated users

Reply:
changed to performDeviceRegistration and added it in ITestBroker. Also added implmentations for this method in each of the brokers